plugins/inputs/statsd/running_stats.go

```diff
@@ -69,12 +77,32 @@ func (rs *RunningStats) AddValue(v float64) {
 		// Reached limit, choose random index to overwrite in the percentile array
 		rs.perc[rand.Intn(len(rs.perc))] = v
 	}
+
+	rs.MedLen = len(rs.med)
+	// Need to sort for median
+	sort.Float64s(rs.med)
```

Comment:
I suggest to sort in the `Median` function, as it will probably be called less often and actually belongs there.